core/block_events.js

```diff
@@ -31,19 +32,26 @@ goog.require('Blockly.utils.xml');
 
 /**
  * Abstract class for a block event.
- * @param {Blockly.Block} block The block this event corresponds to.
+ * @param {!Blockly.Block=} opt_block The block this event corresponds to.
```

Comment:
Does this mean that we are requiring that if a block is not passed, then we need to pass undefined, instead of null? I believe currently we either pass null or undefined.
This case (when block is not passed) happens often for `Blockly.Events.UI` (since some UI events do not have an associated block, but block is the first param for the constructor).

Reply:
We were currently only allowing Block or null for Block events. With the switch to make everything optional, I don't see a need for allowing null there. Null was being used to signal a blank event, and that has been taken care of my making everything optional.

In the case of UI events, we still allow `null`, as null for the block parameter means that the event is not associated with a block. See jsdoc for UI events:
```
 * @param {?Blockly.Block=} opt_block The affected block.  Null for UI events
 *     that do not have an associated block.  Undefined for a blank event.
```

Reply:
I see, I got confused because I mistook this for the `Blockly.Events.Abstract` class. That makes sense, thanks for clarifying.